src/solvers/flattening/boolbv_map.cpp

```diff
@@ -20,27 +20,18 @@ Author: Daniel Kroening, kroening@kroening.com
 
 std::string boolbv_mapt::map_entryt::get_value(const propt &prop) const
 {
+  if(!is_set)
+    return std::string(literal_map.size(), '*');
+
   std::string result;
 
   result.reserve(literal_map.size());
 
-  for(std::size_t i=0; i<literal_map.size(); i++)
+  for(const auto &literal : literal_map)
   {
-    char ch='*';
+    const tvt value = prop.l_get(literal);
 
-    if(literal_map[i].is_set)
-    {
-      tvt value=prop.l_get(literal_map[i].l);
-
-      if(value.is_true())
-        ch='1';
-      else if(value.is_false())
-        ch='0';
-      else
-        ch='?';
-    }
-
-    result=result+ch;
+    result += (value.is_true() ? '1' : (value.is_false() ? '0' : '?'));
```

Comment:
With the way this is written right now I'd expect something like `10?001` to be possible result, but from the commit message it should either be all `?`s or none of them should be right?

Reply:
Not quite: the bits not being set would be printed as a string of '*'. The '?' refers to an individual variable not having a value under a particular assignment computed by a solver. That said: it seems the above is dead code, so perhaps the distinction between `is_set` (which perhaps should be "is_unconstrained" (!)) and is-not-set-under-an-assignment isn't all that important.